src/org/intellij/erlang/inspection/ErlangUnresolvedFunctionInspection.java

```diff
@@ -80,10 +83,35 @@ private void inspect(PsiElement what, ErlangQAtom target, @Nullable PsiReference
         }
 
         ErlangFunctionReference r = (ErlangFunctionReference) reference;
-        if (r.getArity() < 0) return; //there is no need to inspect incomplete/erroneous code
+        int arity = r.getArity();
+        String name = r.getName();
+        if (arity < 0 || name == null) return;
+
+        String functionPresentation = ErlangPsiImplUtil.createFunctionPresentation(name, arity);
+        String fixMessage = "Create Function " + functionPresentation;
+
         LocalQuickFix[] qfs = PsiTreeUtil.getNextSiblingOfType(what, ErlangModuleRef.class) != null ?
-          new LocalQuickFix[]{} : new LocalQuickFix[]{new ErlangCreateFunctionQuickFix(r.getName(), r.getArity())};
-        registerProblem(holder, target, "Unresolved function " + "'" + r.getSignature() + "'", qfs);
+          new LocalQuickFix[]{} : new LocalQuickFix[]{new ErlangCreateFunctionQuickFix(target, name, arity, fixMessage)};
+
+        registerProblem(holder, target, "Unresolved function " + functionPresentation, qfs);
+      }
+    };
+  }
+
+  private static FunctionTextProvider<ErlangFunctionCallExpression> createProviderForFunctionCall(final ErlangFunctionCallExpression o) {
```

Comment:
This final keyword is dangerous here. If you had another final variable, which you'd use in anonymous class below, this parameter would have been captured, too. Essentially, you'd store a reference to this PSI element in your quickfix. Be careful when using final. 

Add nullability annotations.